src/ParallelAlgorithms/Interpolation/Callbacks/ObserveTimeSeriesOnSurface.hpp

```diff
@@ -89,19 +91,19 @@ auto make_reduction_data(const db::DataBox<DbTags>& box, double time,
 /// see InterpolationTarget for a description of InterpolationTargetTag.
 template <typename TagsToObserve, typename ObservationType,
           typename InterpolationTargetTag>
-struct ObserveTimeSeriesOnSurface {
+struct ObserveTimeSeriesOnSurface
+    : tt::ConformsTo<intrp::protocols::PostInterpolationCallback> {
   using observed_reduction_data_tags = observers::make_reduction_data_tags<
       tmpl::list<typename detail::reduction_data_type<TagsToObserve>::type>>;
   using observation_types = tmpl::list<ObservationType>;
 
   static constexpr double fill_invalid_points_with =
       std::numeric_limits<double>::quiet_NaN();
 
-  template <typename DbTags, typename Metavariables>
-  static void apply(
-      const db::DataBox<DbTags>& box,
-      Parallel::GlobalCache<Metavariables>& cache,
-      const typename InterpolationTargetTag::temporal_id::type& temporal_id) {
+  template <typename DbTags, typename Metavariables, typename TemporalId>
+  static void apply(const db::DataBox<DbTags>& box,
+                    Parallel::GlobalCache<Metavariables>& cache,
+                    const TemporalId& temporal_id) {
```

Comment:
I'm unsure about this change, and suggest you drop it, along with the similar changes below (to `SendGhWorldtubeData` etc). This change shouldn't be needed to conform to the protocol. As I commented above, perhaps the right thing to do is to constrain the compile-time interface more, not less, but no need to decide in this PR.

Reply:
See other comment explaining how templates are used to deduce types in protocols.